From f1320ed41990cd359492399b565bfd5480e07ec5 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Thu, 15 Dec 2022 08:55:02 +0200
Subject: [PATCH] [SW-112776] preboot: i2ci: fix device status

Change device status to 'recovery', in order to correctly reflect the
state that the OAM is in.

Change-Id: Ie2c58063fa2aaa2f239ea153ab6a450a76c10c50
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index e0cb065..2c6250b 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -17,6 +17,9 @@
 
 #include <uart.h>
 
+#include <hl_i2c_indirect_protocol.h>
+#include <gaudi2/i2c_indirect_protocol.h>
+
 static struct cmd_module_ops cmd_ops[] = {
 	{	/* PCI module */
 		.setup = module_pci_setup,
@@ -80,6 +83,11 @@
 	device_init_pll();
 }
 
+static void set_oam_recovery_status(void)
+{
+	op_status.device_status = HL_I2CI_REQ_DEV_STS_RECOVERY;
+}
+
 void erom_main_function(void)
 {
 	device_init();
@@ -123,6 +131,8 @@
 	/* In Preboot, module LKD is responsible for setting the below flag */
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 
+	set_oam_recovery_status();
+
 	preboot_puts("In recovery mode\n");
 
 	module_handle_cmds(cmd_ops, ARRAY_SIZE(cmd_ops));
diff --git a/include/preboot.h b/include/preboot.h
index 3dba9af..3b4417b 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -101,6 +101,7 @@
 extern uintptr_t _start;	/* Start of ppboot/preboot region */
 extern uintptr_t __stack_top;	/* End of ppboot/preboot region */
 extern u32 __stack;		/* stack last cell value */
+extern struct hl_i2ci_device_status op_status; /* Contains OAM status info */
 
 #define STACK_CANARY_MAGIC_WORD	0x8BADF00D
 
diff --git a/lib/preboot/i2cslave_indirect.c b/lib/preboot/i2cslave_indirect.c
index a6ee266..552c39d 100644
--- a/lib/preboot/i2cslave_indirect.c
+++ b/lib/preboot/i2cslave_indirect.c
@@ -102,7 +102,7 @@
 static struct i2csi_wdata_read_context req_rd;
 static struct i2csi_wdata_write_context req_wd;
 
-static struct hl_i2ci_device_status op_status = {
+struct hl_i2ci_device_status op_status = {
 	.device_status = HL_I2CI_REQ_DEV_STS_PREBOOT,
 	.valid_opcode_map = SUPPORTED_OPCODES_MASK,
 };
